src/mmrm.cpp

```diff
@@ -56,17 +57,18 @@ Type objective_function<Type>::operator() ()
   Type sum_log_det = 0.0;
   // Get theta_size
   int theta_size = theta.size() / n_groups;
-
+  // Get the spatial indicator
+  bool spatial = is_spatial(cov_type);
```

Comment:
you can directly pass this from R? then you don't need to define helper function